python/tests/operations/join.py

```diff
@@ -0,0 +1,272 @@
+import unittest
+from os import path
+
+import numpy as np
+from numpy.testing import assert_equal
+
+from equistore import Labels, TensorBlock, TensorMap
+from equistore.io import load
+from equistore.operations.allclose import allclose_raise
+from equistore.operations.join import join
+from equistore.operations.slice import slice
+
+
+DATA_ROOT = path.join(path.dirname(__file__), "..", "data")
+
+
+class TestJoinTensorMap(unittest.TestCase):
+    def setUp(self):
+        self.ps = load(path.join(DATA_ROOT, "qm7-power-spectrum.npz"), use_numpy=True)
+        self.se = load(
+            path.join(DATA_ROOT, "qm7-spherical-expansion.npz"), use_numpy=True
+        )
+        self.first_block = self.ps.block(0)
+
+        keys_first_block = Labels(
+            names=self.ps.keys.names,
+            values=np.array(self.ps.keys[0].tolist()).reshape(1, -1),
+        )
+
+        self.ps_first_block = TensorMap(keys_first_block, [self.first_block.copy()])
+
+        block_extra_grad = self.first_block.copy()
+        gradient = self.ps.block(0).gradient("positions")
+        block_extra_grad.add_gradient(
+            "foo", gradient.data, gradient.samples, gradient.components
+        )
+        self.ps_first_block_extra_grad = TensorMap(keys_first_block, [block_extra_grad])
+
+    def test_single_tensormap(self):
+        """Test error raise if only one tensormap is provided."""
+        with self.assertRaises(ValueError) as err:
+            join(self.ps, axis="properties")
+        assert "provide at least two" in str(err.exception)
```

Comment:
```suggestion
        self.assertIn("provide at least two", str(err.exception))
```

unfortunately unittest does not use the default assert syntax